presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -190,7 +190,13 @@ public OptionalInt tryConvertToDirect(int maxDirectBytes)
 
         if (inRowGroup) {
             directColumnWriter.beginRowGroup();
-            writeDictionaryRowGroup(dictionaryValues, rowGroupValueCount, values);
+            if (!writeDictionaryRowGroup(dictionaryValues, rowGroupValueCount, values, maxDirectBytes)) {
+                directColumnWriter.finishRowGroup();
```

Comment:
Is this supposed to be here? I thought you would need this only if the switch to direct worked.

Reply:
It depends on whether row group has to be finished before close can be called. From other usage of this API, this doesn't seem to be a requirement. I will remove this.